### PR TITLE
Fix Alpine Linux support

### PR DESCRIPTION
**Pull request by thrix**
The previous patches did not update the `prepare` install plugin. Add a functional test to verify the alpine image is really testable.

Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage

### CODE REVIEW

File: tests/provision/container/alpine/test.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            execute --how tmt --script whoami" 2
+        rlAssertGrep "fail: Bash is required on the guest." $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartTest "Test alpine with bash"
+        rlRun -s "tmt run -vv --all --id $run --verbose --scratch \
+            provision --how container --image localhost/alpine:tmt-tests \
+            execute --how tmt --script whoami" 0
+        rlAssertGrep "out: root" $rlRun_LOG
+    rlPhaseEnd
```
Comment (happz) [inline tests/provision/container/alpine/test.sh:26]: I'd recommend adding a test (or a phase) for the `prepare/install` plugin, with at least one `require` package - that was the issue, right?

File: tmt/steps/provision/__init__.py [hunk 2]
```diff
@@ -190,6 +190,8 @@ def _execute(
         except tmt.utils.RunError as exc:
             if exc.stdout and 'Please login as the user' in exc.stdout:
                 raise tmt.utils.GeneralError(f'Login to the guest failed.\n{exc.stdout}') from exc
+            if exc.stderr and 'executable file `/bin/bash` not found' in exc.stderr:
+                raise tmt.utils.GeneralError('Bash is required on the guest.') from exc
```
Comment (happz) [inline tmt/steps/provision/__init__.py:194]: I'd refer to `tmt.utils.DEFAULT_SHELL` instead of `/bin/bash` and `Bash`.

Comment (thrix) [reply tmt/steps/provision/__init__.py:194]: yeah, did not see it

File: tests/provision/container/alpine/test.sh (hunk 1 above)
Comment (thrix) [reply tests/provision/container/alpine/test.sh:26]: yeah, true true

Comment (happz) [reply tests/provision/container/alpine/test.sh:26]: I plan to add this test case to `tests/prepare/install`, among other tests checking various distros, so I suppose you can ignore this particular suggestion.

**Review by happz — APPROVED** [review]

**Review by psss — DISMISSED** [review]
Thanks for the fix! Looks good, just two comments.

File: plans/main.fmf [hunk 3]
```diff
@@ -3,12 +3,18 @@ provision:
...
   - how: install
     package:
       - jq
+      - make
       - python3-pip
 
+  # Make sure test container images are built
+  - how: shell
+    script:
+      - make -C $TMT_TREE images-unit-tests
```
[the remark this answers is not in the document]
Comment (happz) [reply plans/main.fmf]: Dropped in 9b277efd. I will refactor the image building later anyway.

File: tests/provision/container/alpine/test.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    rlPhaseStartTest "Test vanilla alpine without bash"
+        rlRun -s "tmt run --all --id $run --verbose --scratch \
+            provision --how container --image docker.io/alpine \
+            execute --how tmt --script whoami" 2
+        rlAssertGrep "fail: /bin/bash is required on the guest." $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartTest "Test alpine with bash"
+        rlRun -s "tmt run -vv --all --id $run --verbose --scratch \
+            provision --how container --image localhost/alpine:tmt-tests \
```
[the remark this answers is not in the document]
Comment (happz) [reply tests/provision/container/alpine/test.sh]: Fixed in 9b277efd.

[the remark this answers is not in the document]
Comment (thrix) [reply tests/provision/container/alpine/test.sh]: ty

**Review by lukaszachy — APPROVED** [review]

**Pull request merged**